### ISSUE TITLE
Script compilation errors when updating connection string through Azure DevOps

### ISSUE THREAD

**Issue by gabrielsmelchor** [otykier/TabularEditor#711]
@otykier  ,

We're getting a script compilation error when calling the script to update the connection string within an Azure DevOps task. We are targeting a model update on a dataset hosted in a Power BI Premium Workspace. We have to issue when deploying the model, just when we call the script. We have tested the script locally within the 'Advanced Scripting' inside of Tabular Editor, and it works.

Error in Azure (all tasks are Powershell Tasks):

`Script compilation errors:
SetConnectionStringFromEnvPQ.cs(1,32): error CS1002: ; expected
PowerShell exited with code '1'.`

Powershell task to deploy called within Azure:

```
`$p = Start-Process -FilePath "TabularEditor.exe" -NoNewWindow -PassThru -Wait -ArgumentList "_$(Build.DefinitionName)\drop\Model.bim -S  `"SetConnectionStringFromEnvPQ.cs`" -D `"$(ASConnectionString)`" `"$(Dataset)`"  -P -V -E -W"`
```

Update connection string script called SetConnectionStringFromEnvPQ.cs (the model has Power Query partitions):

``` // This script is used to replace the server and database names across
// all power query partitions, with the ones provided through environment
// variables:
// This will be an environment variable
var newDatabase = "\"" + Environment.GetEnvironmentVariable("db") + "\"";
// This will always be the same
var enquireServer = "\"" + Environment.GetEnvironmentVariable("server") + "\"";
// This function will extract all quoted values from the M expression, returning a list of strings
// with the values extracted (in order), but ignoring any quoted values where a hashtag (#) precedes
// the quotation mark:
var split = new Func<string, List<string>>(m => {
    var result = new List<string>();
    var i = 0;
    foreach (var s in m.Split('"'))
    {
        if (s.StartsWith("DW") && i % 2 == 0) i = -2;
        if (i >= 0 && i % 2 == 1) result.Add(s);
        i++;
    }
    return result;
});
var GetServer = new Func<string, string>(m => split(m)[0]);
var GetDatabase = new Func<string, string>(m => split(m)[1]);  
// Loop through all partitions on the model, replacing the server and database names from the partitions
// with the ones specified in environment variables:
    foreach (var p in Model.AllPartitions.OfType<MPartition>())
    {
        // Added the try loop because not all partitions have the same string format of <server> <db> and the function would fail on GetDatabase
        try {
          var oldServer = GetServer(p.Expression);
var oldDatabase = GetDatabase(p.Expression);
          // Only update partition if it is pointing to enquiredw
            if (oldServer == enquireServer) {
                  oldDatabase.Output();
                  oldServer.Output();
                  // Code below this actually changes the partition string
                  p.Expression = p.Expression.Replace(oldDatabase, newDatabase);
                  p.Expression.Output();
              } ;
          } 
        // This will instruct the loop to continue to the next partition
          catch 
          { return;
          }
    }

We have confirmed that the environment variables are returning the expected strings within Azure. As the script works locally (with hardcoding the server and database variable), we're wondering if it's an issue with calling the script in the model deployment.

Any guidance would be greatly appreciated!

**Comment by gabrielsmelchor** [comment]
// This script is used to replace the server and database names across
// all power query partitions, with the ones provided through environment
// variables:

// This will be an environment variable
var newDatabase = "DW132";
// This will always be the same
var enquireServer = "enquiredwpool.database.windows.net";
var server = "\"" + enquireServer + "\"";

// This function will extract all quoted values from the M expression, returning a list of strings
// with the values extracted (in order), but ignoring any quoted values where a hashtag (#) precedes
// the quotation mark:
var split = new Func<string, List<string>>(m => {
    var result = new List<string>();
    var i = 0;
    foreach (var s in m.Split('"'))
    {
        if (s.StartsWith("DW") && i % 2 == 0) i = -2;
        if (i >= 0 && i % 2 == 1) result.Add(s);
        i++;
    }
    return result;
});
var GetServer = new Func<string, string>(m => split(m)[0]);
var GetDatabase = new Func<string, string>(m => split(m)[1]);  



// Loop through all partitions on the model, replacing the server and database names from the partitions
// with the ones specified in environment variables:
    foreach (var p in Model.AllPartitions.OfType<MPartition>())
    {
        // Added the try loop because not all partitions have the same string format of <server> <db> and the function would fail on GetDatabase
        try {
          var oldServer = GetServer(p.Expression);
          var oldDatabase = GetDatabase(p.Expression);
          // Only update partition if it is pointing to enquiredw
            if ( oldServer == enquireServer) {
                  // Code below this actually changes the partition string
                  p.Expression = p.Expression.Replace(oldDatabase, newDatabase);
              } ;
          } 
        // This will instruct the loop to continue to the next partition
          catch 
          { return;
          }
    }

**Comment by otykier** [comment]
Tabular Editor will try to locate a file with the provided name after the `-S` switch. If no such file exists, it assumes that you have provided an inline script, which is of course not the case here, and so you get this c# syntax error: `; expected`.

Could you try to provide the full path of the file after `-S` switch? I'm guessing it would be something like:

`"_$(Build.DefinitionName)\drop\SetConnectionStringFromEnvPQ.cs`"

...of course depending on where in your repository the file is located.

**Comment by gabrielsmelchor** [comment]
@otykier  thank you! We should have reviewed the documentation further before posting here. We have successfully updated the connection string on release. BTW I think we're a testament to Tabular Editor. We came into this knowing nothing about: Powershell, C#, or Azure. However, we've put together a CI/CD solution for 400+ clients using your tool!

Thanks for your contribution to the community.
